qrisk/stats.py

```diff
@@ -62,33 +64,35 @@ def cum_returns(returns, starting_value=None):
     -----
     For increased numerical accuracy, convert input to log returns
     where it is possible to sum instead of multiplying.
+    PI((1+r_i)) - 1 = exp(ln(PI(1+r_i)))     # x = exp(ln(x))
+                    = exp(SIGMA(ln(1_r_i))   # ln(a*b) = ln(a) + ln(b)
     """
 
     # df_price.pct_change() adds a nan in first position, we can use
-    # that to have cum_returns start at the origin so that
+    # that to have cum_logarithmic_returns start at the origin so that
     # df_cum.iloc[0] == starting_value
     # Note that we can't add that ourselves as we don't know which dt
     # to use.
     if pd.isnull(returns.iloc[0]):
         returns.iloc[0] = 0.
 
-    df_cum = np.exp(np.log(1 + returns).cumsum())
+    df_cum = np.exp(np.log1p(returns).cumsum())
 
-    if starting_value is None:
+    if starting_value == 0:
         return df_cum - 1
```

Comment:
You can't because if any of the returns is 0.0, then the result will be 0.0 overall. [.05, 0.0,] would yield 0. Adding 1 to the entire series allows the result to be 0.5, but forces the 1 to be subtracted at the end.